Minor text tweaks to overview, pulled in architecture and evaluation … (#108)
opea-project/docs: examples/ChatQnA/ChatQnA_Guide.rst

.. note:: This guide is in its early development and is a work-in-progress with
   placeholder content.

> Overview

Reviewer: ```suggestion
Overview DON"T ACCEPT THIS SUGGESTION
```

Architecture
************

The ChatQnA architecture is displayed below:

> ![alt text](image.png)

Reviewer: Not sure what this is for, delete it.
```suggestion
```

.. figure:: /GenAIExamples/ChatQnA/assets/img/chatqna_flow_chart.png
   :alt: ChatQnA Architecture Diagram

> .. code-block:: python

Reviewer: What graph drawing language is this?  It looks a bit like graphviz, but syntax isn't correct.  If it were, you could use the graphviz directive instead of a code block:

```
.. graphviz::
```

Reviewer: I'm having issues getting this to render as a mermaid diagram myself.  Let's remove the diagram for now so we can get this document merged (without that diagram).

> .. code-block:: python

Reviewer: ```suggestion
.. mermaid::
```
And then everything that's the mermaid graph description below needs to be indented three spaces

> graph LR

Reviewer: Need to indent to keep withing code-block directive.
```suggestion
   graph LR
```

        RER([Rerank 'TEI'<br>8000])
       LLM([LLM 'text-generation TGI'<br>9000])
    end

    direction TB
    TEI_EM{{TEI embedding service<br>8090}}
    VDB{{Vector DB<br>8001}}
    %% Vector DB interaction
    TEI_EM -.->|d|VDB

    DP([OPEA Data Preparation<br>6007])
    LLM_gen{{TGI/vLLM/ollama Service}}

    direction TB
    RER([OPEA Reranking<br>8000])
    TEI_RER{{TEI Reranking service<br>8808}}

    subgraph User Interface
        direction TB
        a[User Input Query]
        Ingest[Ingest data]
        UI[UI server<br>Port: 5173]
    end

    subgraph ChatQnA GateWay
        direction LR
        GW[ChatQnA GateWay<br>Port: 8888]
    end

    %% Data Preparation flow
    %% Ingest data flow
    direction LR
    Ingest[Ingest data] -->|a| UI
    UI -->|b| DP
    DP -.->|c| TEI_EM

    %% Questions interaction
    direction LR
    a[User Input Query] -->|1| UI
    UI -->|2| GW
    GW ==>|3| ChatQnA-MegaService
    EM ==>|4| RET
    RET ==>|5| RER
    RER ==>|6| LLM

    %% Embedding service flow
    direction TB
    EM -.->|3'| TEI_EM
    RET -.->|4'| TEI_EM
    RER -.->|5'| TEI_RER
    LLM -.->|6'| LLM_gen

    subgraph Legend

>         X([Micsrservice])

Reviewer 2: ```suggestion
        X([Microservice])
```

        Y{{Service from industry peers}}
        Z[Gateway]
    end

Microservice Outline and Diagram
================================

> OPEA projects typcially consist of a collection of microservices to create a megaservice, all facilitated by a gateway. A microservice is a component designed to perform a specific function or task. Microservices are building blocks, offering the fundamental services. Microservices promote the modularity, flexibility, and scalability of the system. A megaservice is a higher-level architectural construct composed of one or more microservices, providing the capability to assemble end-to-end applications.

Reviewer 2: ```suggestion
A GenAI application or pipeline in OPEA  typically consists of a collection of microservices to create a megaservice, accessed via a gateway. A microservice is a component designed to perform a specific function or task. Microservices are building blocks, offering the fundamental services. Microservices promote modularity, flexibility, and scalability in the system. A megaservice is a higher-level architectural construct composed of one or more microservices, providing the capability to assemble end-to-end applications.
```

> OPEA projects typcially consist of a collection of microservices to create a megaservice, all facilitated by a gateway. A microservice is a component designed to perform a specific function or task. Microservices are building blocks, offering the fundamental services. Microservices promote the modularity, flexibility, and scalability of the system. A megaservice is a higher-level architectural construct composed of one or more microservices, providing the capability to assemble end-to-end applications.

Reviewer: It's a good practice to keep line length around 80 characters so you don't have these really long lines in Git to deal with.

> The gateway serves as the interface for users to access. Gateway routing incoming requests to the appropriate Microservices within the Megaservice architecture. See `GenAI Components <https://github.com/opea-project/GenAIComps>`_for more information.

Reviewer 2: ```suggestion
The gateway serves as the interface for users to access. The gateway routes incoming requests to the appropriate microservices within the megaservice. See `GenAI Components <https://github.com/opea-project/GenAIComps>`_for more information.
```

Reviewer: If you make Malini's suggested edit (or if you don't) you need a space between the underscore and for:
``<https://github.com/opea-project/GenAIComps>`_ for``

Reviewer: please make this edit to add a space
```suggestion
The gateway serves as the interface for users to access. Gateway routing incoming requests to the appropriate Microservices within the Megaservice architecture. See `GenAI Components <https://github.com/opea-project/GenAIComps>`_ for more information.
```
